client/packages/common/src/ui/components/inputs/DatePickers/BaseDatePickerInput/BaseDatePickerInput.tsx

```diff
@@ -17,22 +17,36 @@ export const BaseDatePickerInput: FC<
   DatePickerProps<Date> & {
     error?: string | undefined;
     width?: number;
+    onBlur?: (e: React.SyntheticEvent) => void;
   }
-> = ({ error, onChange, width, ...props }) => {
+> = ({ error, onChange, onError, width, onBlur, ...props }) => {
   const theme = useAppTheme();
-  const [validationError, setValidationError] = React.useState<string | null>(null);
+  const [internalError, setInternalError] = React.useState<string | null>(null);
   const t = useTranslation('common');
 
-return (
+  return (
     <DatePicker
       slots={{
         textField: TextField,
       }}
-      onChange={(date,context) => {
-        const {validationError} = context;
+      onChange={(date, context) => {
+        const { validationError } = context;
 
-        setValidationError(validationError ? t(`error.date_${validationError}` as LocaleKey, {defaultValue: validationError}) : null);
-        onChange?.(date, context);
+        if (validationError) {
+          if (onError) onError(validationError, date);
+          else
+            setInternalError(
+              validationError
+                ? t(`error.date_${validationError}` as LocaleKey, {
+                    defaultValue: validationError,
+                  })
```

Comment:
with this block now used in three places it would be better to have it as a method - it was a bit of a hacky suggestion to start with sorry and I feel bad about it being used in diff places.
Would be better as a long form, less flexible approach so that we're not playing fast and loose with translation typings:

```
setInternalError(formattedDateError(t, validationError));
```

with a shared helper fn:

```
const formattedDateError = (t: TypedTFunction<LocaleKey>, error: string) => {
  if (!error) return null;

  switch (error) {
    case 'date_invalidDate':
      return t('error.date_invalidDate');
    case 'date_minDate':
      return t('error.date_minDate');
    case 'date_maxDate':
      return t('error.date_maxDate');
    case 'date_disablePast':
      return t('error.date_disablePast');
    case 'date_disableFuture':
      return t('error.date_disableFuture');
    default:
      return error;
  }
};
```